### ISSUE TITLE
write user stories to specify the format of API endpoints

### ISSUE THREAD

**Issue by abrahmasandra** [uchicago-cs/chigame#114]
We need a shared team documentation (which is also useful for external users) for the format of API endpoints. We need to locate all api endpoints in the api app, thus having all api endpoints start with site/api/…

This can be defined through user stories in our documentation. 

For example,

“Vi can visit site/api/users/ and then see a django rest framework generated view that displays the users of chigame”
“Arnav can visit site/api/users/[username] to see a django rest framework generated view with information about a specific user by entering their username”

**Comment by Esterello2** [comment]
I started on some API endpoint user stories:
                                                                          
# POST User Endpoint:
-Unregistered user Alex will create a unique username and password combination in the initial Chigame "Sign Up" page along with a verified email. This will POST a new user object in the User table of the Chigame database.
# PATCH User Endpoint:
-Existing user Von wants to edit his information such as profile picture or email, and can do so in the "Update Profile" settings of the home page for a signed-in Chigame page.
#  POST Tournament Endpoint:
-Admin user Charlie wants to create a public tournament with a max cap of 32 players for a Checkers game. He specifies the number of players, the game ID for Checkers from the game library, tournament style, and visibility (public vs. private) in the POST request to the tournaments database.

-Users will automatically get a notification of this open public tournament through an automated GET API request of the newly added tournament where they will be able to see what game is being played, how many users have joined the tournament, and if there is still availability.

**Comment by bsokolow10** [comment]
As an avid Chigame player, John is excited that he can access to API’s of the game and wants to use them to prototype his own version of Chigame. To do this, he can go to site/api/documentation (or site/api_documentation?) to find a description of what specific information he can access on how to interpret them (for user management, tournament registration, lobby creation, and gameplay, etc.)

**Comment by abrahmasandra** [comment]
@Esterello2 I think your user stories are really good. I only think that for the POST/PATCH User endpoints, you shouldn't be saying that a user goes to a "Update Profile" or "Sign Up" page. I am assuming that since this is from an API, the user wouldn't be interacting with the HTML of the webpage, and would instead just be sending POST/PATCH requests with JSON data. I may be wrong though.

**Comment by abrahmasandra** [comment]
@bsokolow10 I like the user story about how to access the documentation. I would just edit the URLs, so that it is the actual URL of the Github wiki page for the API portion of the chigame project.

**Comment by Esterello2** [comment]
I added POST and GET friends endpoints in the user stories: 
# PATCH User Endpoint:
-Existing user Von wants to edit his information such as profile picture or email, and can do so in the "Update Profile" settings of the home page for a signed-in Chigame page.

# POST Friends:
-Existing user Devi accepts a "Friend" invite from another user Allen. Allen's and Devi's "Friends" are updated accordingly, where they are now linked with their respective User IDs in the Friend's database.

**Comment by cuyakonwu** [comment]
I think this issue is extremely broad, we've broken this one down several times into other issues and will continually add more user stories as we create/ideate new endpoints

**Comment by abrahmasandra** [comment]
Closing Comment:

In this issue, we brainstormed different User stories for different potential endpoints. They have now been copied onto the `wiki` page for the `API` team. Additionally, we broke this issue down into multiple other issues, and added more user stories within those new issues. 

@Esterello2  @bsokolow10: wrote some User stories as seen in comments above
@abrahmasandra : commented on User stories/gave suggestions 
@cuyakonwu : helped split large issue into multiple smaller issues
